### DISCUSSION TITLE
Cart error (only on local development)

### DISCUSSION THREAD

**Discussion by roarschouten** [EvgeniyMukhamedjanov/liquid-ajax-cart#42] [Q&A]
Hi, I just wanted to change some minor things to an existing store I build. But there is an issue with the cart and I hope you can help. I have never faced this problem before and did not change anything.

The specific problem I am facing is that the "floating cart" opens perfectly but the products are not loaded (so the cart stays empty), when the page gets refreshed the product shows up in the cart. Also, the delete and quantity buttons are not working anymore. The weird thing is when I push the theme to Shopify everything works fine. So this issue only appears in local development mode.

No error shows up in the console until I press the delete button in the cart. (see screenshot)

<img width="1778" alt="image" src="https://user-images.githubusercontent.com/62511295/190406799-cee69f91-bac6-47b7-9f53-8c01a7bef06c.png">

The thing is I use your library for multiple projects, and I even copied all the elements from another project (where the cart works fine in dev mode), but still no success.

I hope you can help, thanks in advance.

**Comment by EvgeniyMukhamedjanov** [comment]
Hm, maybe you switched between diferent markets/languages/currencies so the Shopify root url changed?

**Reply by EvgeniyMukhamedjanov** [reply]
Can you please show to what url the request is sent when you click on a quantity buttons in the Chrome inspector network tab and what is your current url (should be http://127.0.0.1:9292/...)

**Reply by roarschouten** [reply]
Thank you for your quick reply! This is the URL when I click on a quantity button: http://127.0.0.1:9292/cart/change.js

<img width="524" alt="image" src="https://user-images.githubusercontent.com/62511295/190451985-917bb278-90cb-439c-bd06-9339aff3f6f7.png">

**Reply by roarschouten** [reply]
It looks like the specific problem is that the cart does not update/refresh after something happens.  (add product, quantity change, etc.) For example, when I add a product to the cart, it opens empty, but when I refresh the page manually, the product is visible in the cart. After that, I can increase or decrease the quantity amount with 1 item and then need to refresh the page and open the cart manually to see the changes.

**Reply by EvgeniyMukhamedjanov** [reply]
I'll try to replicate it on my local environment. I was told few times about this bug but couldn't replicate. 
Do you use Shopify markets/different language versions/different currencies on your website that change that base use from just website.com to website.com/eu for example?

**Reply by roarschouten** [reply]
The website runs on a subdomain: shop.website.com.

It is a single language store, and they do not use different currencies. I just checked the apps, and there is one app called "Geolocation" installed. I just deleted it, but the issue is still there, unfortunately. I have run the theme on another Shopify store, and then the problem is gone. So it looks like something bugs on this specific Shopify store.

The following apps are installed.

- Facebook•Sales channel
- GDPR/CCPA + Cookie Management (I deleted this one as well to check if it solves the issue, but no success)
- Klaviyo: Email Marketing & SMS
- Online Store•Sales channel
- Picqer (Warehouse software)

**Reply by EvgeniyMukhamedjanov** [reply]
Hey, I've just faced with the same issue on one of websites I'm working with.
I'll try to figure out what is wrong there and let you know!

**Reply by edoardolunardi** [reply]
Looks like I'm facing a very similar issue here #94 and here #95

**Reply by EvgeniyMukhamedjanov** [reply]
Hey @edoardolunardi 
Is this happening only on local environment for you?

**Reply by edoardolunardi** [reply]
Hey @EvgeniyMukhamedjanov no, it's happening in a theme preview as well. Unfortunately due to NDA I cannot share it :( but I tried to be as much specific as possible in my 2 issues.

**Reply by EvgeniyMukhamedjanov** [reply]
@edoardolunardi 
I'll try to take a look at it during this week.
